src/SonarScanner.MSBuild/Properties/AssemblyInfo.cs

```diff
@@ -19,7 +19,15 @@
  */
 
 using System.Reflection;
+using System.Runtime.CompilerServices;
 
 [assembly: AssemblyTitle("SonarScanner.MSBuild")]
 [assembly: AssemblyProduct("SonarScanner.MSBuild")]
 [assembly: AssemblyDescription("")]
+
+// FIXME: add public key
+#if SignAssembly
```

Comment:
Educational: what is this about?

Reply:
`[InternalsVisibleTo]` is needed by [this change](https://github.com/SonarSource/sonar-scanner-msbuild/pull/1789/files?diff=split&w=1#diff-f01a53d1a0753af5e850b3353d383448d2ef4fbd0653950a359b4101e5be76d4R47), that was done since the second constructor is only there for testing, and should only be visible to testing methods. I didn't want to have a `public /*for testing*/` constructor.

In an untrustable environment, where assemblies can only be trusted when signed, `SonarScanner.MSBuild` should "trust" `SonarScanner.MSBuild.Test` only when the second is signed with a recognized public key. Otherwise, any dll could pretend to be `SonarScanner.MSBuild.Test`, and being granted without any further check the access to `internal` methods.

Reply:
@costin-zaharia-sonarsource  Do you know how I can get the public key of the signed version of `SonarScanner.MSBuild.Test.dll`. The same as in the code above is done [here](https://github.com/SonarSource/sonar-scanner-msbuild/blob/36b091efec2f7c236c4f98eadc90576ac851953f/src/SonarScanner.MSBuild.PreProcessor/Properties/InternalsVisibleTo.cs) and [here](https://github.com/SonarSource/sonar-scanner-msbuild/blob/36b091efec2f7c236c4f98eadc90576ac851953f/src/SonarScanner.MSBuild.TFS/Properties/InternalsVisibleTo.cs), for example. 

If it's too much of a hassle, I can:
- either revert the visibility change, from `internal` to `public` (but we may just push the problem to the next time)
- or diverging from the other projects, and not condition access to internal members by public key when `SignAssembly` is defined. Given that internal members can be accessed anyway via reflection or .NET 8 zero-cost private access, it seems an unnecessary protection.

Reply:
@cristian-ambrosini-sonarsource  Discussed offline with @costin-zaharia-sonarsource:
- `PublicKey` in `[InternalsVisibleTo]` refers to the public key used for strong naming of the assembly. It is the same public key used for all the other DLLs. It's not a signature of the other assembly, as I wrongly assumed above. If it was, by the way, any change to the test library would require a change of value, which, while secure, would be very inconvenient.
- Without `PublicKey`, the `[InternalVisibleTo]` would not work. See [here](https://www.meziantou.net/internalsvisibleto-with-strong-named-assemblies.htm) for more details.
- The issue, however, doesn't currently appear in the CI since the signature only appears on the master CI pipeline. Temporarily changing `IS_RELEASE_BRANCH` in the azure pipeline configuration to the value `true` would reveal the problem. I am going to try that on another branch.
- Anyway, I have pushed a commit to set the same `PublicKey` as all other DLLs. That should fix the issue.